library/std/src/canon.qs

```diff
@@ -246,6 +247,24 @@ namespace Microsoft.Quantum.Canon {
         return snd;
     }
 
+    /// # Summary
+    /// Exchanges the labels of two qubits. This can function like a SWAP gate, but without needing to
+    /// perform any operations on the qubits themselves. The relabeling is done purely in qubit ID management.
+    ///
+    /// # Input
+    /// ## q0
+    /// The first qubit to swap.
+    /// ## q1
+    /// The second qubit to swap.
+    ///
+    /// # Remarks
+    /// This operation is useful when you need to swap qubits in a way that does not incur any quantum operations.
+    /// Note that when compiling for execution on hardware with limited qubit connectivity, this operation
+    /// may not result in any changes to qubit adjacency and a SWAP gate may still be required.
+    operation SwapLabels(q0 : Qubit, q1 : Qubit) : Unit {
```

Comment:
Some possible name ideas:

- `Relabel`
- `VirtualSwap`
- `LogicalSwap`
- `NameSwap`
- `NominalSwap`

Reply:
Based on feedback, I adjusted the approach to expose `Relabel` as the more generic operation for handling arbitrary permutations.